### ISSUE TITLE
Build failing for ghc-8.4.1

### ISSUE THREAD

**Issue by harendra-kumar** [zalora/getopt-generics#56]
See [this travis job](https://travis-ci.org/harendra-kumar/unicode-transforms/jobs/354593558):

```
src/WithCli/Flag.hs:19:10: error:
    • Could not deduce (Semigroup (Flag a))
        arising from the superclasses of an instance declaration
      from the context: Monoid a
        bound by the instance declaration at src/WithCli/Flag.hs:19:10-36
    • In the instance declaration for ‘Monoid (Flag a)’
   |
19 | instance Monoid a => Monoid (Flag a) where
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
cabal: Leaving directory '/tmp/cabal-tmp-4524/getopt-generics-0.13.0.1'
```

**Comment by soenkehahn** [comment]
@harendra-kumar: I've created #57, it'd be great if you could take a look.

**Comment by soenkehahn** [comment]
Closed via #57.
